app/src/main/java/com/brandon/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,291 @@
+                            createAccountState.value = CreateAccountUiState.Loaded
+                        }
+                    }
+                }
+            } else {
+                createAccountState.value = CreateAccountUiState.Idle(
+                    isEmailError = validMail,
+                    isLocationError = validLoc,
+                    isDateError = validDob,
+                    email = email,
+                    firstName = firstName,
+                    lastName = lastName,
+                    location = location,
+                    dateOfBirth = date,
+                )
+            }
+        }
+        Log.e(VIEWMODEL_ERROR_TAG, "${createAccountState.value}")
+    }
+
+    private fun isValidLocation(location: String): Boolean {
+        return location.length <= LOCATION_CHAR_LIMIT
+    }
+
+    private fun isValidEmail(email: String): Boolean {
+        return PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
+    }
+
+    private fun isValidDateOfBirth(dob: String): Boolean {
+        dob.forEach {
+            if (!it.isDigit()) {
+                return false
+            }
+        }
+        return try {
+            val inputFormat = SimpleDateFormat("MMddyyyy")
+            val tempDate = inputFormat.parse(dob)
+            val outputFormat = SimpleDateFormat("yyyy-MM-dd")
+            val dateString = tempDate?.let { outputFormat.format(it) }
+            sqlDate = Date.valueOf(dateString)
+            true
+        } catch (e: Exception) {
+            Log.e(VIEWMODEL_ERROR_TAG, "Invalid Date of Birth Provided.")
+            false
+        }
+    }
+
+    private fun isValidCode(code: String): Boolean {
+        return code.isDigitsOnly()
+    }
+
+    fun isNetworkAvailable(context: Context?): Boolean {
+        if (context == null) return false
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            val capabilities =
+                connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+            if (capabilities != null) {
+                return true
+            }
+        } else {
+            val activeNetworkInfo = connectivityManager.activeNetworkInfo
+            if (activeNetworkInfo != null && activeNetworkInfo.isConnected) {
+                return true
+            }
+        }
+        return false
+    }
+
+    class DateTransformation() : VisualTransformation {
+        override fun filter(text: AnnotatedString): TransformedText {
+            return dateFilter(text)
+        }
+
+        private fun dateFilter(text: AnnotatedString): TransformedText {
+
+            val trimmed = if (text.text.length >= 8) text.text.substring(0..7) else text.text
+            var out = ""
+            for (i in trimmed.indices) {
+                out += trimmed[i]
+                if (i % 2 == 1 && i < 4) out += "/"
```

Comment:
Is this level of manual string creation needed here? No date methods to do this? This seems risky

Reply:
will remove,